src/python/dxpy/scripts/dx.py

```diff
@@ -1967,12 +1973,15 @@ def head(args):
     if args.lines > 0:
         try:
             if handler._class == 'file':
-                handler._read_bufsize = 1024*32
-                for line in handler:
-                    print(line)
-                    counter += 1
-                    if counter == args.lines:
-                        break
+                try:
+                    handler._read_bufsize = 1024*32
+                    for line in handler:
+                        print(line)
+                        counter += 1
+                        if counter == args.lines:
+                            break
+                except UnicodeDecodeError:
+                    sys.stdout.write("File contains binary data")
```

Comment:
I was thinking we could add such a nicer printout for Py2 but let's not focus of the old (that's going away hopefully) too much!